app/services/cache/MongoCacheClient.scala

```diff
@@ -214,16 +242,23 @@ class MongoCacheClient(appConfig: ApplicationConfig, db: () => DefaultDB, applic
   /**
     * Fetches the whole cache and returns default where not exists
     */
-  def fetchAllWithDefault(credId: String): Future[Cache] =
+  def fetchAllWithDefault(credId: String): Future[Cache] = {
     fetchAll(Some(credId)).map {
       _.getOrElse(Cache(credId, Map.empty))
     }
+  }
 
   /**
     * Removes the item with the specified id from the cache
     */
-  def removeById(credId: String) =
-    collection.delete().one(key(credId)) map handleWriteResult
+  def removeById(credId: String):Future[Boolean] = {
+    collection.findOneAndDelete(key(credId)).toFuture()
+      .map { result => true
+      }
+      .recover { case _ => false }
```

Comment:
⛏️ Should we log this error or just not handle?

Reply:
As we are removing, the only fail case I guess is that the element we want to remove doesn't exist, in which case I guess that's the goal either way.

We could always log this result but I'm unsure if this is needed? 🤔

Reply:
Logging in case there's something wrong with the connection or Mongo schema has changed (recover will fire if it can't parse the result for example).